compiler/src/dotty/tools/dotc/transform/PatternMatcher.scala

```diff
@@ -839,7 +839,7 @@ object PatternMatcher {
               default
           }
         case ResultPlan(tree) =>
-          if (tree.symbol eq defn.throwMethod) tree // Namely MatchError
+          if (tree.tpe <:< defn.NothingType) tree // Namely MatchError
```

Comment:
The comment is not accurate anymore. It should say "for example MatchError". Otherwise LGTM.